syntax = "proto3";

/**
A message representing a book in a library.
 */
message Book {
    // The ISBN of the book.
    string isbn = 1;
    // The title of the book.
    string title = 2;
    // The author of the book.
    Author author = 3;
    // The genre of the book.
    Genre genre = 4;
    // The year the book was published.
    int32 published_year = 5;
}

/**
A message representing an author.
 */
message Author {
    // The name of the author.
    string first_name = 1;
    // The last name of the author.
    string last_name = 2;
    // The year the author was born.
    int32 birth_year = 3;
}

enum Genre {
    UNKNOWN = 0;
    FICTION = 1;
    NON_FICTION = 2;
    SCIENCE = 3;
}